crates/rome_js_analyze/src/analyzers/nursery/no_misused_new.rs

```diff
@@ -0,0 +1,232 @@
+            })
+            .to_owned(),
+            RuleState::InterfaceMisusedConstructor(_) => (markup! {
+                "Do not use the `constructor` method in interface."
+            })
+            .to_owned(),
+            RuleState::TypeAliasMisusedConstructor(_) => (markup! {
+                "Do not use the `constructor` method in type alias."
+            })
+            .to_owned(),
+        }
+    }
+
+    fn note(&self) -> MarkupBuf {
+        match self {
+            RuleState::ClassMisusedNew(_) => (markup! {
+                "`new` is not typical in class."
+            })
+            .to_owned(),
+            RuleState::InterfaceMisusedNew(_) => (markup! {
+                "`new` is misleading in interface."
+            })
+            .to_owned(),
+            RuleState::InterfaceMisusedConstructor(_) => (markup! {
+                "Interface shouldn't have `constructor`."
+            })
+            .to_owned(),
+            RuleState::TypeAliasMisusedConstructor(_) => (markup! {
+                "Type alias can't use `constructor`."
+            })
+            .to_owned(),
+        }
+    }
+
+    fn range(&self) -> TextRange {
+        match self {
+            RuleState::ClassMisusedNew(range)
+            | RuleState::InterfaceMisusedNew(range)
+            | RuleState::InterfaceMisusedConstructor(range)
+            | RuleState::TypeAliasMisusedConstructor(range) => *range,
+        }
+    }
+}
+
+impl Rule for NoMisusedNew {
+    type Query = Ast<DeclarationQuery>;
+    type State = RuleState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        match node {
+            DeclarationQuery::TsInterfaceDeclaration(decl) => {
+                let interface_ident = decl.id().ok()?.name_token().ok()?;
+                for member in decl.members() {
+                    match member {
+                        AnyTsTypeMember::TsConstructSignatureTypeMember(construct) => {
+                            if construct.new_token().ok().is_none() {
+                                continue;
+                            }
+                            let any_ts_type = construct.type_annotation()?.ty().ok()?;
+                            let reference_type = any_ts_type.as_ts_reference_type()?;
+                            let return_type_ident = extract_return_type_ident(&reference_type)?;
+                            if interface_ident.text_trimmed() == return_type_ident.text_trimmed() {
+                                return Some(RuleState::InterfaceMisusedNew(construct.range()));
+                            }
+                        }
+                        AnyTsTypeMember::TsMethodSignatureTypeMember(method) => {
+                            let method_name = method.name().ok()?;
+                            let method_name = method_name.as_js_literal_member_name()?.syntax();
```

Comment:
To simplify the code, we could implement a method `AnyJsClassMemberName::name` with a similar implementation to [`AnyJsObjectMemberName::name`](https://github.com/biomejs/biome/blob/main/crates/rome_js_syntax/src/expr_ext.rs#L991).